components/Usertable.js

```diff
@@ -0,0 +1,324 @@
+            value={values.inputLastName}
+            onChange={handleChange}
+            inputProps={{
+              name: "inputLastName"
+            }}
+          />
+          <TextField
+            margin="dense"
+            id="email"
+            label="Email Address"
+            type="email"
+            fullWidth
+            value={values.inputEmail}
+            onChange={handleChange}
+            inputProps={{
+              name: "inputEmail"
+            }}
+          />
+          <TextField
+            margin="dense"
+            id="password"
+            label="Password"
+            type="password"
+            fullWidth
+            value={values.inputPassword}
+            onChange={handleChange}
+            inputProps={{
+              name: "inputPassword"
+            }}
+          />
+          <FormControl fullWidth style={{ marginTop: 10 }}>
+            <InputLabel>User Type</InputLabel>
+            <Select
+              value={values.userType}
+              onChange={handleChange}
+              inputProps={{
+                name: "userType"
+              }}
+              autoWidth
+            >
+              <MenuItem value={0}>Admin</MenuItem>
+              <MenuItem value={1}>Member</MenuItem>
+            </Select>
+          </FormControl>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleAddClose} color="primary">
+            Cancel
+          </Button>
+          <Button onClick={handleAddSubmit} color="primary">
+            Save
+          </Button>
+        </DialogActions>
+      </Dialog>
+
+      <Paper style={style.Paper}>
+        <Grid container>
+          <Grid item sm>
+            <Typography variant="h4">User Management</Typography>
+          </Grid>
+          <Grid item sm>
+            <div style={{ textAlign: "right" }}>
+              <Button
+                variant="contained"
+                color="primary"
+                size="medium"
+                style={{ marginRight: 20 }}
+                onClick={handleInviteClick}
+              >
+                Invite User
+              </Button>
+              <Button
+                variant="contained"
+                color="primary"
+                size="medium"
+                onClick={handleAddClick}
+              >
+                Add User
+              </Button>
+            </div>
+          </Grid>
+        </Grid>
+      </Paper>
+      <MaterialTable
+        title="Registered Users"
+        columns={state.columns}
+        style={{ padding: 20, margin: 20 }}
+        data={state.data}
+        editable={{
+          onRowUpdate: (newData, oldData) =>
+            new Promise(resolve => {
+              setTimeout(() => {
+                resolve();
+                const data = [...state.data];
+                data[data.indexOf(oldData)] = newData;
+                setState({ ...state, data });
+              }, 600);
+            }),
```

Comment:
Out of curiosity - why are you using a Promise and timeout here? Was this in a guide that you can link me?